apis/cloudflare/src/proxy.ts

```diff
@@ -65,31 +72,59 @@ export async function handleProxyV1(
 
   const cache = await caches.open("apikey:cache");
 
+  const credentialsCache = {
+    async get<T>(key: string): Promise<T | null> {
+      const response = await cache.match(apiCacheKey(key));
+      if (response) {
+        return (await response.json()) as T;
+      } else {
+        return null;
+      }
+    },
+    async set<T>(key: string, value: T, { ttl }: { ttl?: number }) {
+      await cache.put(
+        apiCacheKey(key),
+        new Response(JSON.stringify(value), {
+          headers: {
+            "Cache-Control": `public${ttl ? `, max-age=${ttl}}` : ""}`,
+          },
+        }),
+      );
+    },
+  };
+
+  let spanLogger: SpanLogger | undefined;
+  let span: Span | undefined;
+  const parentHeader = request.headers.get(BT_PARENT);
+  if (parentHeader) {
+    let parent;
+    try {
+      parent = resolveParentHeader(parentHeader);
+    } catch (e) {
+      return new Response(
+        `Invalid parent header '${parentHeader}': ${e instanceof Error ? e.message : String(e)}`,
+        { status: 400 },
+      );
+    }
+    span = startSpan({
+      state: await cachedLogin({
+        appUrl: braintrustAppUrl(env).toString(),
+        headers: request.headers,
+        cache: credentialsCache,
+      }),
+      type: "llm",
+      name: "LLM",
```

Comment:
not blocking but i wonder if there's utility in being able to control the name of this span via headers as well. ill leave it to you to since im not too familiar with the practices we use for span names elsewhere and what we typically see customers doing

Reply:
That is a pretty neat idea. We can always make it configurable if it comes up